test/test_decoding.py

```diff
@@ -0,0 +1,144 @@
+import tensorflow as tf
+
+from wavenet import WaveNet
+from test_encoding import manual_encode
+import json
+import numpy as np
+
+def manual_decode(signal, channels):
+    # Calculate inverse mu-law companding and dequantization
+    mu = channels
+    output = signal.astype(np.float32)
+    y = (2*output - 1) / mu
+    x = np.sign(y) * (np.exp(y * np.log(1 + mu)) - 1) / mu
+
+    return x
+
+
+class TestDecode(tf.test.TestCase):
+
+    def setUp(self):
+        quantization_steps=256
+        self.net = WaveNet(batch_size = 1,
+                           channels = quantization_steps,
+                           dilations= [1, 2, 4, 8, 16, 32, 64, 128, 256,
+                                       1, 2, 4, 8, 16, 32, 64, 128, 256],
+                           filter_width=2,
+                           residual_channels = 16,
+                           dilation_channels = 16)
+
+
+    def testUniformRandomNoise(self):
+        # Input: uniformly distributed random noise
+        with self.test_session():
+            # Audio encoding parameters
+            number_of_samples = 512
+            mu = self.net.channels
+
+            # Random seed for repeatability of test
+            np.random.seed(40)
+
+            # Generate audio input for one channel
+            audio = np.random.uniform(-1, 1, number_of_samples)
+
+            # Encode and decode audio input 
+            encoded_audio_manual = manual_encode(audio, mu)
+            decoded_audio_manual = manual_decode(encoded_audio_manual, mu)
+
+            decoded_audio_net_tensor = self.net.decode(encoded_audio_manual)
+            decoded_audio_net = decoded_audio_net_tensor.eval()
+
+            # Shapes should be the same
+            self.assertAllEqual(decoded_audio_net.shape, decoded_audio_manual.shape)
```

Comment:
There's no need to test the shapes, as `assertAllClose` does this automatically.